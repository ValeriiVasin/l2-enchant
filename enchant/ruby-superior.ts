import { createEnchantMap } from '../helpers/create-enchant-map';
import { itemsRange } from '../helpers/items-range';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const rubySuperiorEnchantMap = createEnchantMap({
  strategy: EnchantStrategy.Drop,
  enchantItem: Item.RubyEnergy,
  chances: [30, 30, 20, 10],
  items: itemsRange(Item.RubySuperior_1, Item.RubySuperior_5),
});
