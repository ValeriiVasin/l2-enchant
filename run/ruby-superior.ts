import { rubySuperiorEnchantMap } from '../enchant/ruby-superior';
import { itemsRange } from '../helpers/items-range';
import { run } from '../helpers/run';
import { Item } from '../items';

run(Item.RubySuperior_1, itemsRange(Item.RubySuperior_2, Item.RubySuperior_5), 100_000, rubySuperiorEnchantMap);
