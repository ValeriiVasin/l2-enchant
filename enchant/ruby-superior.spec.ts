import { rubySuperiorEnchantMap } from './ruby-superior';

it('enchant map: ruby superior', () => {
  expect(rubySuperiorEnchantMap).toMatchSnapshot();
});
