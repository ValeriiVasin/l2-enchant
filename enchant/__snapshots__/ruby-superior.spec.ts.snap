// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: ruby superior 1`] = `
Map {
  "Рубин Исключительного Качества Ур.1" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Рубин Исключительного Качества Ур.1",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Рубин Исключительного Качества Ур.2",
      },
      Object {
        "chance": 70,
        "item": "Рубин Исключительного Качества Ур.1",
      },
    ],
  },
  "Рубин Исключительного Качества Ур.2" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Рубин Исключительного Качества Ур.2",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Рубин Исключительного Качества Ур.3",
      },
      Object {
        "chance": 70,
        "item": "Рубин Исключительного Качества Ур.1",
      },
    ],
  },
  "Рубин Исключительного Качества Ур.3" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Рубин Исключительного Качества Ур.3",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Рубин Исключительного Качества Ур.4",
      },
      Object {
        "chance": 80,
        "item": "Рубин Исключительного Качества Ур.1",
      },
    ],
  },
  "Рубин Исключительного Качества Ур.4" => Object {
    "cost": "Энергия Самоцвета",
    "item": "Рубин Исключительного Качества Ур.4",
    "results": Array [
      Object {
        "chance": 10,
        "item": "Рубин Исключительного Качества Ур.5",
      },
      Object {
        "chance": 90,
        "item": "Рубин Исключительного Качества Ур.1",
      },
    ],
  },
}
`;
